storage/test/files.test.js

```diff
@@ -56,425 +59,461 @@ function getSample () {
-      var filesSample = getSample();
 
-      filesSample.sample.copyFile(bucketName, undefined, undefined, undefined, function (err) {
-        assert(err);
-        assert(err.message = '"srcFileName" is required!');
-      });
-    });
-    it('should require destBucketName', function () {
-      var filesSample = getSample();
-
-      filesSample.sample.copyFile(bucketName, fileName, undefined, undefined, function (err) {
-        assert(err);
-        assert(err.message = '"destBucketName" is required!');
-      });
-    });
-    it('should require destFileName', function () {
-      var filesSample = getSample();
-
-      filesSample.sample.copyFile(bucketName, fileName, bucketName, undefined, function (err) {
-        assert(err);
-        assert(err.message = '"destFileName" is required!');
-      });
-    });
     it('should handle error', function () {
-      var error = 'copyFileError';
-      var filesSample = getSample();
-      filesSample.mocks.file.move = sinon.stub().callsArgWith(1, error);
-
-      filesSample.sample.copyFile(bucketName, fileName, bucketName, fileName, function (err) {
-        assert.equal(err, error);
-      });
-    });
-  });
-  describe('printUsage', function () {
-    it('should print usage', function () {
-      var filesSample = getSample();
-
-      filesSample.sample.printUsage();
-
-      assert(console.log.calledWith('Usage: node files COMMAND [ARGS...]'));
-      assert(console.log.calledWith('\nCommands:\n'));
-      assert(console.log.calledWith('\tlist BUCKET_NAME'));
-      assert(console.log.calledWith('\tlistByPrefix BUCKET_NAME PREFIX [DELIMITER]'));
-      assert(console.log.calledWith('\tupload BUCKET_NAME FILE_NAME'));
-      assert(console.log.calledWith('\tdownload BUCKET_NAME SRC_FILE_NAME DEST_FILE_NAME'));
-      assert(console.log.calledWith('\tdelete BUCKET_NAME FILE_NAME'));
-      assert(console.log.calledWith('\tgetMetadata BUCKET_NAME FILE_NAME'));
-      assert(console.log.calledWith('\tmakePublic BUCKET_NAME FILE_NAME'));
-      assert(console.log.calledWith('\tmove BUCKET_NAME SRC_FILE_NAME DEST_FILE_NAME'));
-      assert(console.log.calledWith('\tcopy BUCKET_NAME SRC_FILE_NAME DEST_BUCKET_NAME DEST_FILE_NAME'));
+      var error = 'error';
+      var sample = getSample();
+      var callback = sinon.stub();
+      var options = {
+        srcBucket: bucketName,
+        srcFile: fileName,
+        destFile: fileName,
+        destBucket: bucketName
+      };
+      sample.mocks.file.copy = sinon.stub().callsArgWith(1, error);
+
+      sample.program.copyFile(options, callback);
+
+      assert(callback.calledOnce, 'callback called once');
+      assert.equal(callback.firstCall.args.length, 1, 'callback received 1 argument');
+      assert(callback.firstCall.args[0], 'callback received error');
+      assert.equal(callback.firstCall.args[0].message, error.message, 'error has correct message');
     });
   });
+
   describe('main', function () {
```

Comment:
I'm assuming we still want to keep this `main()` test around, even with `yargs` - or is that something we should discuss?